### PR TITLE
tensorflow plugin implementation

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Tensorflow plugin for Flyte that leverages https://github.com/kubeflow/tf-operator to run distributed tensorflow jobs on k8s.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Tensorflow plugin for Flyte that leverages https://github.com/kubeflow/tf-operator to run distributed TensorFlow jobs on k8s.
https://pypi.org/project/flyteidl/0.17.36/
Related PRs:
TFOperator support in FlytePlugins
https://github.com/lyft/flyte/issues/115

Subtask:
https://github.com/lyft/flyteidl/pull/71
https://github.com/lyft/flyteplugins/pull/103
https://github.com/lyft/flyte/pull/424

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
@pingsutw can I look at it tomorrow morning?

**Comment by pingsutw** [comment]
@kumare3 Sure, I'm not in a hurry.

**Comment by kumare3** [comment]
the PR is failing because of some un-related error. I see that happening for all PR's

**Comment by kumare3** [comment]
This actually looks pretty good to me. We should probably add an example in flytesnacks.

**Comment by kumare3** [comment]
@wild-endeavor do you have any comments?

**Comment by pingsutw** [comment]
@kumare3 Thanks for the review.
After PR is merged, I will add an example in flytesnacks.

**Comment by wild-endeavor** [comment]
@pingsutw I don't think I have access to your fork.  Could you help us clean up the lint errors?  If not, feel free to change this to a PR to go in something other than the master branch, and I can pull it and modify it. 

Thanks!

**Comment by wild-endeavor** [comment]
Hey I want to make a few more changes to this PR, so merged it into a different branch, but on the main fork.  I'll open up a new PR into master and tag you on it.  Thanks @pingsutw !@
